library/main.py

```diff
@@ -0,0 +1,125 @@
+from fastapi import Depends, FastAPI, HTTPException
+from sqlalchemy.orm import Session
+
+from library import crud, models, schemas
+from library.database import SessionLocal, engine
+
+models.Base.metadata.create_all(bind=engine)
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(
+        author: schemas.AuthorCreate,
+        db: Session = Depends(get_db)
+) -> models.DBAuthor:
+    db_author_name = crud.get_author_by_name(db=db, author_name=author.name)
+    if db_author_name:
+        raise HTTPException(status_code=404, detail="Author already exists")
+    db_author = crud.create_author(db, author)
+    return db_author
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_authors(
+        skip: int = 0,
+        limit: int = 100,
+        db: Session = Depends(get_db)
+) -> list[models.DBAuthor]:
+    authors = crud.get_authors(db=db, skip=skip, limit=limit)
+    return authors
+
+
+@app.get("/authors/{author_id}/", response_model=schemas.Author)
+def read_author(
+        author_id: int,
+        db: Session = Depends(get_db)
+) -> models.DBAuthor:
+    db_author = crud.get_author(db=db, author_id=author_id)
+    if db_author is None:
+        raise HTTPException(status_code=404, detail="Author not found")
+    return db_author
+
+
+@app.put("/authors/{author_id}/", response_model=schemas.Author)
+def update_author(
+        author_id: int,
+        author: schemas.AuthorCreate,
+        db: Session = Depends(get_db)
+) -> models.DBAuthor:
+    db_author = crud.update_author(db=db, author_id=author_id, author=author)
+    if db_author is None:
+        raise HTTPException(status_code=404, detail="Author not found")
+    return db_author
+
+
+@app.delete("/authors/{author_id}/", response_model=schemas.Author)
+def delete_author(
+        author_id: int,
+        db: Session = Depends(get_db)
+) -> models.DBAuthor:
+    db_author = crud.delete_author(db=db, author_id=author_id)
+    if db_author is None:
+        raise HTTPException(status_code=404, detail="Author not found")
+    return db_author
```

Comment:
do you really need to return something after delete? It is up to you btw, just asking)